modules/aws/ec2-files.go

```diff
@@ -15,85 +16,170 @@ type RemoteFileSpecification struct {
 	RemotePathToFileFilter map[string][]string //A map of the files to fetch, where the keys are directories on the remote host and the values are filters for what files to fetch from the directory. The filters support bash-style wildcards.
 	UseSudo                bool
 	SshUser                string
-	KeyPair                *Ec2Keypair
+	SshOptions             *SshOptions
 	LocalDestinationDir    string //base path where to store downloaded artifacts locally. The final path of each resource will include the ip of the host and the name of the immediate parent folder.
 }
 
+// Specify one of KeyPair, SshAgent, or OverrideSshAgent should be specified
+// Any more (or less) than one will result in error on use
+type SshOptions struct {
+	// Username
+	UserName string
+	// Port
+	Port uint16
+
+	// EC2 Key Pair
+	KeyPair *Ec2Keypair
+	// If true, use default SSH agent on local system (started externally, available at SSH_AUTH_SOCK)
+	SshAgent bool
+	// Override SSH agent on local system
+	OverrideSshAgent *ssh.SshAgent
+
+	// The enabled authentication method
+	// Internally will be one of "keypair", "sshagent", or "overridesshagent"
+	enabledAuthMethod string
+}
+
+func (s *SshOptions) Validate() error {
+	if s.UserName == "" {
+		return fmt.Errorf("SshOptions: UserName field cannot be empty")
```

Comment:
Please define custom, exported error types for these errors so it's possible for others to check the type of an error when handling it (rather than relying solely on string error messages).

Reply:
interesting thought, and yea I can do that. would make testing easier also so I can see the value